### PR TITLE
Issue with Polymer/paper-elements

### PR DESCRIPTION
**Pull request by george-aprozeanu**
Hi,

I've ran into an issue with Polymer/paper-elements which depends on the whole list of components for polymer. However, itself doesn't have a main file which resolves, and the code skips it's dependency list.

I've commented out the return and it correctly pushes out the rest of the dependencies for me.

### CODE REVIEW

**Comment by stephenplusplus** [comment]
@CWSpear @ruyadorno what do you think about not requiring a component to have "main" defined before including its dependencies mains?

**Comment by CWSpear** [comment]
Hmm... that's mighty tricky as I would think that *most* cases, it *is* an issue if the base component (i.e. `paper-elements`) doesn't have a `main`, and we shouldn't just silently go about as if all was well.

Maybe allow the functionality (i.e. remove the `return`) but display a warning?

Or perhaps, can we handle these edge cases by using an `override` of an empty main and change the `dep.main.length === 0` check to something like `_.has(dep, 'main')`?

**Comment by stephenplusplus** [comment]
@george-aprozeanu can you link to a specific bower package that doesn't define main? I'm poking around https://github.com/PolymerElements/ and found https://github.com/PolymerElements/paper-progress/blob/master/bower.json for example, which has a main.

Still need to fully think this one through, and would love to get more opinions.

// ping @addyosmani for the polymer stuff -- is it common for bower.json's to not define a `main`?

**Comment by george-aprozeanu** [comment]
https://github.com/PolymerElements/paper-elements/blob/master/bower.json

@stephenplusplus if you look in the bower it does mention a main, but its not physically there in the repo. I must have explained it wrong, its not about an undefined main but a wrong main.

However in both cases I think it should work by grabbing the deps. There seems to be nothing wrong with how they defined paper elements, its sort of a meta package.

**Comment by stephenplusplus** [comment]
Sorry, I'm catching on now.

I'm wondering why they include a `main` if it's a file that doesn't exist?

Either way, I think I agree that we should include dependencies regardless.

File: lib/detect-dependencies.js [hunk 1]
```diff
@@ -175,8 +175,10 @@ function gatherInfo(config) {
 
     if (dep.main.length === 0 && !depIsExcluded) {
       // can't find the main file. this config file is useless!
-      config.get('on-main-not-found')(component);
-      return;
+      onfig.get('on-main-not-found')(component);
```
Comment (stephenplusplus) [inline lib/detect-dependencies.js]: `onfig` -> `config`

File: lib/detect-dependencies.js [hunk 2]
```diff
@@ -175,8 +175,10 @@ function gatherInfo(config) {
...
+      // Polymer/paper-elements comes with a missing main but has useful dependencies
+      // skipping the return correctly includes the transient dependencies
+      // return;
```
Comment (stephenplusplus) [inline lib/detect-dependencies.js]: You can just remove all the comments.

Comment (addyosmani) [inline lib/detect-dependencies.js]: bower.json: We've been a little inconsistent here. I'm talking to the team
about getting our house in order. Sorry about that!

**Comment by stephenplusplus** [comment]
Fixed in 6fb88e0160274babba15a969b398a64d27750bda  - thanks!

**Pull request closed without merging**